aws/ecs_cluster.go

```diff
@@ -0,0 +1,132 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+const firstSeenTagKey = "cloud-nuke-first-seen"
+
+// Tag an ECS cluster identified by the given cluster ARN with a tag that has the given key and value
+func tagEcsCluster(awsSession *session.Session, clusterArn *string, tagKey string, tagValue string) error {
+	svc := ecs.New(awsSession)
+	input := &ecs.TagResourceInput{
+		ResourceArn: clusterArn,
+		Tags: []*ecs.Tag{
+			{
+				Key:   aws.String(tagKey),
+				Value: aws.String(tagValue),
+			},
+		},
+	}
+
+	_, err := svc.TagResource(input)
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+	return nil
+}
+
+func getAllEcsClustersOlderThan(awsSession *session.Session, region string, excludeAfter time.Time) ([]*string, error) {
+	awsSession, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region),
+	})
+
+	clusterArns, err := getAllEcsClusters(awsSession)
+	if err != nil {
+		logging.Logger.Errorf("Error getting all ECS clusters")
+		return nil, err
+	}
+
+	var filteredEcsClusters []*string
+	for _, clusterArn := range clusterArns {
+
+		firstSeenTime, err := getClusterTag(awsSession, clusterArn, firstSeenTagKey)
+		if err != nil {
+			logging.Logger.Errorf("Error getting the `first_seen` tag for ECS cluster with ARN %s", *clusterArn)
+			return nil, err
+		}
+
+		if firstSeenTime.IsZero() {
+			err := tagEcsCluster(awsSession, clusterArn, firstSeenTagKey, time.Now().UTC().String())
+			if err != nil {
+				logging.Logger.Errorf("Error tagigng the ECS cluster with ARN %s", *clusterArn)
+				return nil, err
+			}
+		}
+
+		if excludeAfter.After(firstSeenTime) {
```

Comment:
Should this be in an `else`? After all, if the cluster was not tagged (the `IsZero()` check above), we'd never want to delete it, would we?